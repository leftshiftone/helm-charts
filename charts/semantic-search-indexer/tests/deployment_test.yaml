suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: default
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.name
          value: RELEASE-NAME-semantic-search-indexer-app
      - matchSnapshot:
          path: spec
  - it: custom values
    set:
      image.tag: latest
      resources.limits.cpu: 1000m
      resources.limits.memory: 1Gi
      app.minioExistingSecret: some-thing
      app.minioExistingSecretKey: someKey
    asserts:
      - matchSnapshot:
          path: spec
