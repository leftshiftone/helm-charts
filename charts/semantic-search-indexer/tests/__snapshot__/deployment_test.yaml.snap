custom values:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: semantic-search-indexer
    strategy:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 0
      type: RollingUpdate
    template:
      metadata:
        annotations:
          checksum/config: 0fd75029fb2303d4b904b01be3e7e46254b806d1da7be06f74860693cfde9cae
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: semantic-search-indexer
          application: semantic-search-indexer
      spec:
        containers:
        - env:
          - name: MINIO_PASSWORD
            valueFrom:
              secretKeyRef:
                key: someKey
                name: some-thing
          envFrom:
          - configMapRef:
              name: RELEASE-NAME-semantic-search-indexer
          image: 007098893018.dkr.ecr.eu-central-1.amazonaws.com/semantic-search-indexer:latest
          imagePullPolicy: IfNotPresent
          livenessProbe:
            httpGet:
              path: /health
              port: http
          name: semantic-search-indexer
          ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          readinessProbe:
            httpGet:
              path: /health
              port: http
          resources:
            limits:
              cpu: 1000m
              memory: 1Gi
          securityContext: {}
          startupProbe:
            failureThreshold: 30
            httpGet:
              path: /health
              port: http
            periodSeconds: 10
        initContainers:
        - command:
          - sh
          - -c
          - until curl --fail --silent --output /dev/null $ELASTICSEARCH_URL/_cluster/health;
            do echo "Waiting for elasticsearch to be available at $ELASTICSEARCH_URL/_cluster/health...";
            sleep 3; done
          env:
          - name: ELASTICSEARCH_URL
            valueFrom:
              configMapKeyRef:
                key: ES_URL
                name: RELEASE-NAME-semantic-search-indexer
          image: curlimages/curl:7.84.0
          imagePullPolicy: IfNotPresent
          name: semantic-search-indexer-elasticsarch-check
        - command:
          - sh
          - -c
          - until curl --fail --silent --output /dev/null http://$MINIO_HOST/minio/health/live;
            do echo "Waiting for elasticsearch to be available at http://$MINIO_HOST/minio/health/live...";
            sleep 3; done
          env:
          - name: MINIO_HOST
            valueFrom:
              configMapKeyRef:
                key: MINIO_HOST
                name: RELEASE-NAME-semantic-search-indexer
          image: curlimages/curl:7.84.0
          imagePullPolicy: IfNotPresent
          name: semantic-search-indexer-minio-check
        securityContext: {}
        serviceAccountName: RELEASE-NAME-semantic-search-indexer
default:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: semantic-search-indexer
    strategy:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 0
      type: RollingUpdate
    template:
      metadata:
        annotations:
          checksum/config: 0fd75029fb2303d4b904b01be3e7e46254b806d1da7be06f74860693cfde9cae
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: semantic-search-indexer
          application: semantic-search-indexer
      spec:
        containers:
        - envFrom:
          - configMapRef:
              name: RELEASE-NAME-semantic-search-indexer
          image: 007098893018.dkr.ecr.eu-central-1.amazonaws.com/semantic-search-indexer:0.0.0
          imagePullPolicy: IfNotPresent
          livenessProbe:
            httpGet:
              path: /health
              port: http
          name: semantic-search-indexer
          ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          readinessProbe:
            httpGet:
              path: /health
              port: http
          resources: {}
          securityContext: {}
          startupProbe:
            failureThreshold: 30
            httpGet:
              path: /health
              port: http
            periodSeconds: 10
        initContainers:
        - command:
          - sh
          - -c
          - until curl --fail --silent --output /dev/null $ELASTICSEARCH_URL/_cluster/health;
            do echo "Waiting for elasticsearch to be available at $ELASTICSEARCH_URL/_cluster/health...";
            sleep 3; done
          env:
          - name: ELASTICSEARCH_URL
            valueFrom:
              configMapKeyRef:
                key: ES_URL
                name: RELEASE-NAME-semantic-search-indexer
          image: curlimages/curl:7.84.0
          imagePullPolicy: IfNotPresent
          name: semantic-search-indexer-elasticsarch-check
        - command:
          - sh
          - -c
          - until curl --fail --silent --output /dev/null http://$MINIO_HOST/minio/health/live;
            do echo "Waiting for elasticsearch to be available at http://$MINIO_HOST/minio/health/live...";
            sleep 3; done
          env:
          - name: MINIO_HOST
            valueFrom:
              configMapKeyRef:
                key: MINIO_HOST
                name: RELEASE-NAME-semantic-search-indexer
          image: curlimages/curl:7.84.0
          imagePullPolicy: IfNotPresent
          name: semantic-search-indexer-minio-check
        securityContext: {}
        serviceAccountName: RELEASE-NAME-semantic-search-indexer
